fe1-web/src/features/linked-organizations/network/LinkedOrgHandler.ts

```diff
@@ -0,0 +1,150 @@
+ */
+export const handleChallengeRequestMessage =
+  (getCurrentLaoId: LinkedOrganizationsConfiguration['getCurrentLaoId']) =>
+  (msg: ProcessableMessage) => {
+    if (
+      msg.messageData.object !== ObjectType.FEDERATION ||
+      msg.messageData.action !== ActionType.CHALLENGE_REQUEST
+    ) {
+      console.warn('handleRequestChallengeMessage was called to process an unsupported message');
+      return false;
+    }
+
+    const makeErr = (err: string) => `challenge/request was not processed: ${err}`;
+
+    const laoId = getCurrentLaoId();
+    if (!laoId) {
+      console.warn(makeErr('no Lao is currently active'));
+      return false;
+    }
+    if (msg.messageData instanceof ChallengeRequest) {
+      const challengeRequest = msg.messageData as ChallengeRequest;
+      if (challengeRequest.timestamp) {
+        return true;
+      }
+    }
+    return false;
+  };
+
+/**
+ * Handles an federationInit message.
+ */
+export const handleFederationInitMessage =
+  (getCurrentLaoId: LinkedOrganizationsConfiguration['getCurrentLaoId']) =>
+  (msg: ProcessableMessage) => {
+    if (
+      msg.messageData.object !== ObjectType.FEDERATION ||
+      msg.messageData.action !== ActionType.FEDERATION_INIT
+    ) {
+      console.warn('handleFederationInitMessage was called to process an unsupported message');
+      return false;
+    }
+
+    const makeErr = (err: string) => `federation/init was not processed: ${err}`;
+
+    const laoId = getCurrentLaoId();
+    if (!laoId) {
+      console.warn(makeErr('no Lao is currently active'));
+      return false;
+    }
+
+    if (msg.messageData instanceof FederationInit) {
+      const federationInit = msg.messageData as FederationInit;
+      if (
+        federationInit.lao_id &&
+        federationInit.public_key &&
+        federationInit.server_address &&
+        federationInit.challenge
+      ) {
+        return true;
+      }
+    }
+    return false;
+  };
+
+/**
+ * Handles an federationExpect message.
+ */
+export const handleFederationExpectMessage =
+  (getCurrentLaoId: LinkedOrganizationsConfiguration['getCurrentLaoId']) =>
+  (msg: ProcessableMessage) => {
+    if (
+      msg.messageData.object !== ObjectType.FEDERATION ||
+      msg.messageData.action !== ActionType.FEDERATION_EXPECT
+    ) {
+      console.warn('handleFederationExpectMessage was called to process an unsupported message');
+      return false;
+    }
+
+    const makeErr = (err: string) => `federation/expect was not processed: ${err}`;
+
+    const laoId = getCurrentLaoId();
+    if (!laoId) {
+      console.warn(makeErr('no Lao is currently active'));
+      return false;
+    }
+
+    if (msg.messageData instanceof FederationExpect) {
+      const federationExpect = msg.messageData as FederationExpect;
+      if (
+        federationExpect.lao_id &&
+        federationExpect.public_key &&
+        federationExpect.server_address &&
+        federationExpect.challenge
+      ) {
+        return true;
```

Comment:
same questions as before

Reply:
Currently, as far as I understood it we don't need to store anything except the Challenge. We discussed in our latest meeting that we need to Broadcast the Federation Result Message, so that the Organizer knows the linking of the Organizations worked, and we only need to store that message then.  If it is fine by you I will do this in the next PR then, so I can merge this before the global meeting.